sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts

```diff
@@ -125,6 +137,10 @@ export class SchemaRegistryClient implements SchemaRegistry {
    * @returns Schema with given ID or undefined if no schema was found with the given ID.
    */
   async getSchemaById(id: string, options?: GetSchemaByIdOptions): Promise<Schema | undefined> {
+    const cached = this.idToSchemaMap.get(id);
+    if (cached !== undefined) {
+      return cached;
+    }
     try {
       const { flatResponse, rawResponse } = await getRawResponse(
         (paramOptions) => this.client.schema.getById(id, paramOptions),
```

Comment:
Similar question about updating the two maps.

Reply:
This one is a bit trickier since we do not have enough information to populate one of the caches, so I went with the simplest approach and updated just `idToSchema` for now until I learn more.

Reply:
Ok I implemented bidirectional caching here as well, see https://github.com/Azure/azure-sdk-for-js/pull/16740#discussion_r682941240